airbyte-integrations/connectors/source-amazon-ads/unit_tests/test_streams.py

```diff
@@ -0,0 +1,306 @@
+import responses
+from airbyte_cdk.models import SyncMode
+from jsonschema import validate
+from source_amazon_ads import SourceAmazonAds
+
+
+def setup_responses(
+    profiles_response=None,
+    campaigns_response=None,
+    adgroups_response=None,
+    targeting_response=None,
+    product_ads_response=None,
+    generic_response=None,
+):
+    responses.add(
+        responses.POST,
+        "https://api.amazon.com/auth/o2/token",
+        json={"access_token": "alala", "expires_in": 10},
+    )
+    if profiles_response:
+        responses.add(
+            responses.GET,
+            "https://advertising-api.amazon.com/v2/profiles",
+            body=profiles_response,
+        )
+    if campaigns_response:
+        responses.add(
+            responses.GET,
+            "https://advertising-api.amazon.com/sd/campaigns",
+            body=campaigns_response,
+        )
+    if adgroups_response:
+        responses.add(
+            responses.GET,
+            "https://advertising-api.amazon.com/sd/adGroups",
+            body=adgroups_response,
+        )
+    if targeting_response:
+        responses.add(
+            responses.GET,
+            "https://advertising-api.amazon.com/sd/targets",
+            body=targeting_response,
+        )
+    if product_ads_response:
+        responses.add(
+            responses.GET,
+            "https://advertising-api.amazon.com/sd/productAds",
+            body=product_ads_response,
+        )
+    if generic_response:
+        responses.add(
+            responses.GET,
+            f"https://advertising-api.amazon.com/{generic_response}",
+            json=[],
+        )
+
+
+def get_all_stream_records(stream):
+    records = stream.read_records(SyncMode.full_refresh)
+    return [r for r in records]
+
+
+def get_stream_by_name(streams, stream_name):
+    for stream in streams:
+        if stream.name == stream_name:
+            return stream
+    raise Exception(f"Expected stream {stream_name} not found")
+
+
+@responses.activate
+def test_streams_profile(test_config, profiles_response):
+    setup_responses(profiles_response=profiles_response)
+
+    source = SourceAmazonAds()
+    streams = source.streams(test_config)
+    assert len(streams) == 17
+    actual_stream_names = {stream.name for stream in streams}
+    expected_stream_names = set(
+        [
+            "profiles",
+            "sponsored_display_campaigns",
+            "sponsored_product_campaigns",
+            "sponsored_product_ad_groups",
+            "sponsored_product_keywords",
+            "sponsored_product_negative_keywords",
+            "sponsored_product_ads",
+            "sponsored_product_targetings",
+            "sponsored_products_report_stream",
+            "sponsored_brands_campaigns",
+            "sponsored_brands_ad_groups",
+            "sponsored_brands_keywords",
+            "sponsored_brands_report_stream",
+        ]
+    )
+    assert not expected_stream_names - actual_stream_names
```

Comment:
[Here](https://github.com/airbytehq/airbyte/pull/5023#discussion_r682294846) I left a comment with all needed instructions and code which you can copypaste. These lines of code test `SourceAmazonAds.streams()` function, so please move them into appropriate file - `test_source.py` and remove these lines from `test_streams_profile()` function. If you created 2 test files, one for testing `source`-connected functions and one for testing `streams`-connected functions, then, please don't mix them together.

Reply:
Done